metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -243,8 +244,18 @@ private void flush(Tuple tickTuple) {
     measurement.setValue(result);
     emit(measurement, tickTuple);
 
+    // Execute the update with the old state
+    Map<String, String> tickUpdate = profileConfig.getTickUpdate();
+    Map<String, Object> state = executor.getState();
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
```

Comment:
Ok, I'm confused.  When we do the update on a per-tuple basis, `executor.assign` is called on every expression that will be updated in the config.  Let's say my expression is `"s" : "STATS_ADD(s, value)"`  Now, obviously, within context there is data from my message (`value`) and message from the existing state (the `s` variable).  In the case of a tick update, I'd like to have available just the variables from the existing state (e.g. the `s` variable in my example above).  How would I go about accomplishing that?  I assumed that `s` would be held in the `executor.getState()`, but maybe not.